### PR TITLE
implement check:spec for construction API

### PR DESCRIPTION
**Pull request by irisZhangCB**
Fixes # .

### Motivation

check:spec for construction API
### How is it tested
Tested in both UTXO and account based chain.
#### UTXO: avalanche
Ran `check:spec --configuration-file conf.json`, conf.json and dsl file are located in [here](https://github.com/ava-labs/avalanche-rosetta/blob/master/rosetta-cli-conf/testnet/config.json) and [here](https://github.com/ava-labs/avalanche-rosetta/blob/master/rosetta-cli-conf/testnet/avalanche.ros) (with prefunded account provided)
- Part of `check:spec` log
Screen Shot 2022-06-20 at 10 48 41 PM

- Part of result
Screen Shot 2022-06-20 at 10 48 52 PM

- Transaction is submitted and can be found in explorer
Screen Shot 2022-06-20 at 10 49 28 PM

#### Account based: Optimism
Screen Shot 2022-06-20 at 11 02 25 PM
Screen Shot 2022-06-20 at 11 02 43 PM

### CODE REVIEW

File: cmd/check_spec.go [hunk 1]
```diff
@@ -174,6 +173,29 @@ func runCheckSpecCmd(_ *cobra.Command, _ []string) error {
 
 	// TODO: more checks
 
+	workflows := Config.Construction.Workflows
+	keypair, _ := keys.GenerateKeypair("secp256k1")
```
Comment (irisZhangCB) [inline cmd/check_spec.go:177]: hardcoded it for now, need injection from somewhere

Comment (shiatcb) [reply cmd/check_spec.go:179]: I remember .ros file has curve_type in its workflow def, can we use it?

File: cmd/check_spec.go [hunk 2]
```diff
@@ -174,6 +173,33 @@ func runCheckSpecCmd(_ *cobra.Command, _ []string) error {
...
+
+	// extract operations from dsl file
+	var broadcast *job.Broadcast
+	for _, workflow := range workflows {
+		if workflow.Name == "transfer" {
+			broadcast, _ = parseTransferFromDSL(ctx, workflow, cs.coordinatorHelper)
+		}
+	}
+	// TODO: before calling preprocess, inject recipient and sender's account into broadcast.Intent
+	cs.coordinatorHelper.Preprocess(
```
Comment (shiatcb) [inline cmd/check_spec.go:194]: I think we need to check the error of .Preprocess function.

**Comment by xiaying-peng** [comment]
One question I have is: do we really need a on-chain txn to do check:spec?

File: cmd/check_spec.go [hunk 3]
```diff
@@ -17,6 +17,13 @@ package cmd
 import (
 	"context"
 	"fmt"
+	"github.com/coinbase/rosetta-cli/pkg/processor"
```
Comment (shrimalmadhur) [inline cmd/check_spec.go:20]: can you run go import here?

File: cmd/check_spec.go [hunk 4]
```diff
@@ -505,6 +528,454 @@ func (cs *checkSpec) getAccount(ctx context.Context) (
 	return acct, blockID, currencies, nil
 }
 
+func (cs *checkSpec) ConstructionPreprocess(ctx context.Context, broadcast *job.Broadcast) (map[string]interface{}, error) {
+	printInfo("validating /construction/preprocess ...\n")
+	output := checkSpecConstructionOutput[constructionPreprocess]
+	defer printInfo("/construction/preprocess validated\n")
+
+	metadata, _, err := cs.coordinatorHelper.Preprocess(
```
[the remark this answers is not in the document]
Comment (irisZhangCB) [reply cmd/check_spec.go:536]: `requiredPublicKeys`, upon testing, even not passing these values to `/metadata`(the last parameter [here](https://github.com/coinbase/rosetta-cli/pull/325/files#diff-7a93ac447003717f471f9c6933abe14e212327a094a1d5f4e046d2e79cab3b1dR561) which is pk is set to null), the request can be successful. Wondering why we use this parameter in the request?

**Pull request closed without merging**